kyaml/fieldmeta/fieldmeta.go

```diff
@@ -76,6 +81,53 @@ func (fm *FieldMeta) Read(n *yaml.RNode) error {
 	return nil
 }
 
+// processShortHand parses the comment for short hand ref and loads schema to fm
+func (fm *FieldMeta) processShortHand(comment string) bool {
+	input := map[string]string{}
+	err := json.Unmarshal([]byte(comment), &input)
+	if err != nil {
+		return false
+	}
+	name := input[shortHandRef]
+	if name == "" {
+		return false
+	}
+
+	// check if setter with the name exists, else check for a substitution
+	// setter and substitution can't have same name in shorthand
```

Comment:
Don't we already prohibit setters and substitutions with the same name?

Reply:
Yes. So only we have to do this check as we don't know if it is setter or substitution. So we check if it is setter or substitution.